src/smt.rs

```diff
@@ -1098,14 +1103,17 @@ impl SparseMerkleTree {
             return Ok((vec![info.current_node], vec![info.structure_pos.into()]));
         }
         if total_data == 1 {
-            let (n, v) = self.update_one_node(&info)?;
-            if !n.is_empty() && !v.is_empty() {
-                return Ok((n, v));
+            match self.update_one_node(&info)? {
+                Some((node, structure)) => {
+                    return Ok((vec![node], vec![structure.into()]));
+                },
+                None => {},
```

Comment:
in case of none, shoulnd't this be error?

Reply:
In case of None is should just pass through like before:
```
if total_data == 1 {
    let (n, v) = self.update_one_node(&info)?;
    if !n.is_empty() && !v.is_empty() {
        return Ok((n, v));
    }
}
```

Reply:
ah ok, that's true